File: cy_unity/Packages/com.droidhang.startup@0.3.6/Runtime/LuaScript/Common/PlayerPrefs.lua
local PlayerPrefs = { }

local CSPlayerPrefs = CS.UnityEngine.PlayerPrefs

---@param key string
---@param value number
function PlayerPrefs:SetInt(key, value)
    CSPlayerPrefs.SetInt(key, value)
    CSPlayerPrefs.Save()
end

---@param key string
---@return number
function PlayerPrefs:GetInt(key)
    return CSPlayerPrefs.GetInt(key)
end

---@param key string
---@param default number
---@return number
function PlayerPrefs:GetIntWithDefault(key, default)
    return CSPlayerPrefs.GetInt(key, default)
end

---@param key string
---@param value string
function PlayerPrefs:SetString(key, value)
    CSPlayerPrefs.SetString(key, value)
    CSPlayerPrefs.Save()
end

---@param key string
---@return string
function PlayerPrefs:GetString(key)
    return CSPlayerPrefs.GetString(key)
end

---@param key string
---@param default string
---@return string
function PlayerPrefs:GetStringWithDefault(key,default)
    return CSPlayerPrefs.GetString(key,default)
end

---@param key string
---@return boolean
function PlayerPrefs:HasKey(key)
    return CSPlayerPrefs.HasKey(key)
end

function PlayerPrefs:SetFloat(key, value)
    CSPlayerPrefs.SetFloat(key, value)
    CSPlayerPrefs.Save()
end

function PlayerPrefs:GetFloat(key)
    return CSPlayerPrefs.GetFloat(key)
end

function PlayerPrefs:GetFloatWithDefault(key, defalut)
    return CSPlayerPrefs.GetFloat(key, defalut)
end

function PlayerPrefs:DeleteKey(key)
    CSPlayerPrefs.DeleteKey(key)
end

function PlayerPrefs:DeleteAll()
    CSPlayerPrefs.DeleteAll()
end

return PlayerPrefs

File: cy_unity/Packages/com.droidhang.foundation@0.2.10/Runtime/PlayerPrefs.cs
namespace DHUnityUtil
{
    public static class PlayerPrefs
    {
        public static IPlayerPrefs Instance { get; set; }

#if UNITY_WEBGL && !UNITY_EDITOR
        public static void SetInt(string key, int value)
        {
            Instance.SetInt(key, value);
        }

        public static int GetInt(string key, int defaultValue = 0)
        {
            return Instance.GetInt(key, defaultValue);
        }

        public static void SetString(string key, string value)
        {
            Instance.SetString(key, value);
        }

        public static string GetString(string key, string defaultValue = "")
        {
            return Instance.GetString(key, defaultValue);
        }

        public static void SetFloat(string key, float value)
        {
            Instance.SetFloat(key, value);
        }

        public static float GetFloat(string key, float defaultValue = 0)
        {
            return Instance.GetFloat(key, defaultValue);
        }

        public static void DeleteAll()
        {
            Instance.DeleteAll();
        }

        public static void DeleteKey(string key)
        {
            Instance.DeleteKey(key);
        }

        public static bool HasKey(string key)
        {
            return Instance.HasKey(key);
        }

        public static void Save()
        {
            Instance.Save();
        }
#else
        public static void SetInt(string key, int value)
        {
            UnityEngine.PlayerPrefs.SetInt(key, value);
        }
        public static int GetInt(string key, int defaultValue = 0)
        {
            return UnityEngine.PlayerPrefs.GetInt(key, defaultValue);
        }
        public static void SetString(string key, string value)
        {
            UnityEngine.PlayerPrefs.SetString(key, value);
        }
        public static string GetString(string key, string defaultValue = "")
        {
            return UnityEngine.PlayerPrefs.GetString(key, defaultValue);
        }
        public static void SetFloat(string key, float value)
        {
            UnityEngine.PlayerPrefs.SetFloat(key, value);
        }
        public static float GetFloat(string key, float defaultValue = 0)
        {
            return UnityEngine.PlayerPrefs.GetFloat(key, defaultValue);
        }
        public static void DeleteAll()
        {
            UnityEngine.PlayerPrefs.DeleteAll();
        }
        public static void DeleteKey(string key)
        {
            UnityEngine.PlayerPrefs.DeleteKey(key);
        }
        public static bool HasKey(string key)
        {
            return UnityEngine.PlayerPrefs.HasKey(key);
        }
        public static void Save()
        {
            UnityEngine.PlayerPrefs.Save();
        }
#endif
        
    }

    public interface IPlayerPrefs
    {
        void SetInt(string key, int value);

        int GetInt(string key, int defaultValue = 0);

        void SetString(string key, string value);

        string GetString(string key, string defaultValue = "");

        void SetFloat(string key, float value);

        float GetFloat(string key, float defaultValue = 0);

        void DeleteAll();

        void DeleteKey(string key);

        bool HasKey(string key);

        void Save();
    }
}